### PR TITLE
feat: observabiltiy-as-code 

### PR DESCRIPTION
**Pull request by sudivate**
# Type of PR

- Code changes
- Documentation changes

## Purpose
- Enhanced Observability by adding below resources for Observability-as-code in Bicep modules
   -  Azure Dashboard to monitor key metrics from Azure Data Factory and Azure SQL Server
   - Azure Log Analytics Workspace to collect monitoring Data
   - Enabled diagnostic settings for Azure Data Factory
   - Azure Monitor alert for Azure Data Factory Pipelines
- Demonstrated Data Quality monitoring using OSS Great Expectations in Standardize and Transformation steps of Data pipeline 
-  Sample Data Quality Azure Monitor workbook
-  Updated Docs relevant to all above changes

## Author pre-publish checklist
- [ ] Added test to prove my fix is effective or new feature works
- [x] No PII in logs
- [x] Made corresponding changes to the documentation

### CODE REVIEW

**Comment by devlace** [comment]
Updated the .NET version to 4.7 to resolve build error

**Comment by devlace** [comment]
@sudivate Running into an error in the Databricks notebook. 
image
Looks like its related to a recent breaking change on jinja's side: https://github.com/pallets/jinja/issues/1626

**Review by devlace — CHANGES_REQUESTED** [review]

File: e2e_samples/parking_sensors/databricks/notebooks/02_standardize.py [hunk 1]
```diff
@@ -61,26 +60,152 @@
...
+        "pipeline_run_id": "pipeline_run_id",
+    },
+    runtime_parameters={"batch_data": parkingbay_sdf},  # Your dataframe goes here
+)
+
+# Define Data Quality metric and run Verification
+# create the suite and get a Validator
+
+expectation_suite_name = "parkingbay_data_exception_suite_basic"
+context.create_expectation_suite(expectation_suite_name=expectation_suite_name, overwrite_existing=True)
```
[the remark this answers is not in the document]
Comment (devlace) [reply e2e_samples/parking_sensors/databricks/notebooks/02_standardize.py:129]: As discussed, we'll keep this simple and keep the creation of the expectation suite in the notebook.

**Review by devlace — APPROVED** [review]

**Comment by sudivate** [comment]
Thank you @devlace.

**Pull request merged**